src/psyclone/tests/dynamo0p3_builtins_test.py

```diff
@@ -2583,12 +2638,14 @@ def test_scalar_int_builtin_error(monkeypatch):
         _, invoke_info = parse(
             os.path.join(BASE_PATH, "16.2_integer_scalar_sum.f90"),
             api="dynamo0.3", distributed_memory=dist_mem)
-        with pytest.raises(ParseError) as excinfo:
-            _ = PSyFactory("dynamo0.3",
-                           distributed_memory=dist_mem).create(invoke_info)
-        assert ("an argument to a built-in kernel must be one of ['gh_field', "
-                "'gh_real'] but kernel " + test_builtin_name.lower() + " has "
-                "an argument of type gh_integer" in str(excinfo))
+        if dist_mem:
```

Comment:
I modified the line 551 in dynamo0p3.py to permit reductions into real scalars only:
        if self._type != "gh_real" and \
instead of
        if self._type not in VALID_SCALAR_NAMES and \